tests/do/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -16,4 +48,42 @@ const envSpecific = {
   }
 };
 
+let x = do {
+  let tmp = f();
+  tmp * tmp + 1;
+};
+
+let x = do {
+  if (foo()) {
+    f();
+  } else if (bar()) {
+    g();
+  } else {
+    h();
+  }
+};
+
+function foo() {
+  return (
+    <nav>
+      <Home />
+      {do {
+        if (loggedIn) {
+          <LogoutButton />;
+        } else {
+          <LoginButton />;
+        }
+      }}
+    </nav>
+  );
+}
+
+(do {
+});
+(do {
+} + 1);
+(1 +
+  do {
+  });
```

Comment:
Around `1 + do {}`, the parentheses aren't needed, because `do` isn't at the start of the statement.

Reply:
IMO we should keep them for consistency reasons. Also, this is a highly improbable case since the result of the expression is thrown away.

Reply:
Currently we don't for objects:
```js
// Input
({} + 1);
(1 + {});

// Output
({} + 1);
1 + {};
```

Since the rules for wrapping objects are the same of do expression, I think that they should behave the same.

> the result of the expression is thrown away.

Not with do expressions :stuck_out_tongue: 
```js
var a = do {
  1 + do { if (x) 1; else 2; };
};
```